### ISSUE TITLE
Try `resolve` before removing manifest when instantiating fails

### ISSUE THREAD

**Issue by aplavin** [fonsp/Pluto.jl#3071]
Whenever Pluto cannot instantiate the env, it removes the manifest completely:
```
Operation failed. Updating registries and trying again...
  exception = 
    Unsatisfiable requirements detected for package Statistics [10745b16]:
     Statistics [10745b16] log:
     ├─possible versions are: 1.11.0 - 1.11.1 or uninstalled
     └─restricted to versions 1.10.0 by an explicit requirement — no versions left
    Updating registry at `~/.julia/registries/General.toml`
Warning: Operation failed. Removing Manifest and trying again...
  exception = 
    Unsatisfiable requirements detected for package Statistics [10745b16]:
     Statistics [10745b16] log:
     ├─possible versions are: 1.11.0 - 1.11.1 or uninstalled
     └─restricted to versions 1.10.0 by an explicit requirement — no versions left
```
I believe the most common scenario when it happens is a Julia version update (like in this example).

This automatic behavior is much more destructive compared to the Julia REPL suggestion of running `]resolve`. Removing manifest leads to up/downgrading of all dependencies, basically whatever the Pkg resolver wants to do from scratch.

Can Pluto at least try `]resolve` first to keep the manifest as long as possible?
And potentially – ask the user for an explicit confirmation before performing the destructive action of dropping the manifest...

**Comment by fonsp** [comment]
[julia 1.11 notebook that fails on julia 1.10.jl.zip](https://github.com/user-attachments/files/17541644/julia.1.11.notebook.that.fails.on.julia.1.10.jl.zip)

**Comment by fonsp** [comment]
Pluto does do `resolve` when loading a notebook, but it looks like resolving a Julia 1.11 environment in 1.10 does not always work. My notebook above had the following issue:

https://github.com/JuliaLang/Pkg.jl/issues/4058

**Comment by fonsp** [comment]
We already try resolve before removing the manifest. This particular issue was caused by https://github.com/JuliaLang/Pkg.jl/issues/4058 which will be fixed in upcoming 1.10 and 1.11 releases
